test/labeling/model/test_label_model.py

```diff
@@ -245,6 +245,38 @@ def test_train_model(self):
         next_loss = label_model.loss_mu().detach().numpy().ravel()[0]
 
         self.assertLessEqual(next_loss, init_loss)
+        with self.assertRaisesRegex(
+            Exception, "Loss is NaN. Consider reducing learning rate"
+        ):
+            label_model.train_model(L, n_epochs=10, lr=1e8)
+
+    def test_optimizer(self):
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.train_model(L, n_epochs=1, optimizer="rmsprop")
+        label_model.train_model(L, n_epochs=1, optimizer="adam")
+        with self.assertRaisesRegex(ValueError, "Did not recognize optimizer option"):
+            label_model.train_model(L, n_epochs=1, optimizer="bad_opt")
+
+    def test_lr_scheduler(self):
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.train_model(L, n_epochs=1, lr_scheduler=None)
+        label_model.train_model(L, n_epochs=1, lr_scheduler="exponential")
+        with self.assertRaisesRegex(
+            ValueError, "Did not recognize lr_scheduler option"
+        ):
+            label_model.train_model(L, n_epochs=1, lr_scheduler="bad_scheduler")
+
+    def test_save_and_load(self):
+        L = np.array([[1, 0, 1], [1, 2, 1]])
+        label_model = self._set_up_model(L)
+
+        label_model.train_model(L, n_epochs=1, lr_scheduler=None)
+        label_model.save("./test_save")
```

Comment:
Use tempfile so there isn't a file left after the test runs. Search the other tests for some examples (sometimes creating a new directory, sometimes a new file).

Reply:
done!